packages/api/endpoints/elasticsearch.js

```diff
@@ -0,0 +1,186 @@
+'use strict';
+
+const router = require('express-promise-router')();
+
+const log = require('@cumulus/common/log');
+
+const mappings = require('../models/mappings.json');
+const { defaultIndexAlias, Search } = require('../es/search');
+
+// const snapshotRepoName = 'cumulus-es-snapshots';
+
+async function createEsSnapshot(req, res) {
+  return res.boom.badRequest('Functionality not yet implemented');
+
+  // *** Currently blocked on NGAP ****
+  // const esClient = await Search.es();
+
+  //let repository = null;
+
+  // try {
+  //   const repository = await esClient.snapshot.getRepository({ repository: snapshotRepoName });
+  // }
+  // catch (err) {
+  //   // Handle repository missing exceptions
+  //   if (!err.message.includes('[repository_missing_exception]')) {
+  //     throw err;
+  //   }
+
+  // TO DO: when permission boundaries are updated
+  // repository = await esClient.snapshot.createRepository({
+  //   repository: snapshotRepoName,
+  //   verify: false,
+  //   body: {
+  //     type: 's3',
+  //     settings: {
+  //       bucket: 'lf-internal',
+  //       region: 'us-east-1',
+  //       role_arn: process.env.ROLE_ARN
+  //     }
+  //   }
+  // });
+  // }
+}
+
+async function reindex(req, res) {
+  let sourceIndex = req.body.sourceIndex;
+  let destIndex = req.body.destIndex;
+  const aliasName = req.body.aliasName || defaultIndexAlias;
+
+  const esClient = await Search.es();
+
+  const alias = await esClient.indices.getAlias({
+    name: aliasName
+  });
+
+  // alias keys = index name
+  const indices = Object.keys(alias);
+
+  if (!sourceIndex) {
+    if (indices.length > 1) {
+      // We don't know which index to use as the source, throw error
+      return res.boom.badRequest(`Multiple indices found for alias ${aliasName}. Specify source index as one of [${indices.sort().join(', ')}].`);
+    }
+
+    sourceIndex = indices[0];
+  } else {
+    const sourceExists = await esClient.indices.exists({ index: sourceIndex });
+
+    if (!sourceExists) {
+      return res.boom.badRequest(`Source index ${sourceIndex} does not exist.`);
+    }
+
+    if (indices.includes(sourceIndex) === false) {
+      return res.boom.badRequest(`Source index ${sourceIndex} is not aliased with alias ${aliasName}.`);
+    }
+  }
+
+  if (!destIndex) {
+    const date = new Date();
+    destIndex = `cumulus-${date.getFullYear()}-${date.getMonth() + 1}-${date.getDate()}`;
+  }
+
+  const destExists = await esClient.indices.exists({ index: destIndex });
+
+  if (destExists) {
```

Comment:
So re-index is only for indexing data to a new destination index? Not for ovewriting data in an existing index?

Reply:
If so, I wonder if `copyIndex` is a clearer name for what it is doing

Reply:
I think that would be confusing, since it's using elasticsearch's `reindex` functionality to do this. It's really for exposing that. It goes to another index so you don't run the risk of messing up your current index if something goes wrong. You can swap back to your original index if needed.

Reply:
OK, that makes sense